components/financial_assistance/app/assets/javascripts/financial_assistance/income.js

```diff
@@ -1,33 +1,40 @@
 function stopEditingIncome() {
+  shouldShowEndDateWarningModal = false;
   // remove disabled style from drivers, instructions, and other income kinds
   $('.driver-question, .driver-question a, .instruction-row, .income, .other-income-kind, .unemployment-income').removeClass('disabled');
   // enable inputs on drivers, instructions, and income kinds - except for submit (save) buttons as this would enable the submit button in the dummy forms, and fake-disabled-inputs as they should never be enabled
   $('.driver-question input, .instruction-row input, .income input:not(":input[type=submit], .fake-disabled-input"), .other-income-kind input:not(":input[type=submit], .fake-disabled-input"), .unemployment-income input:not(":input[type=submit], .fake-disabled-input")').removeAttr('disabled');
   
-    // enable "Add New" income buttons
-  $('a.new-income').removeClass('hide');
-  $('#new-unemployment-income, #new-ai-an-income, .add_new_other_income_kind').removeAttr('disabled');
+  // enable "Add New" income buttons
+  $('#new-unemployment-income, #new-ai-an-income, .new-income, .add_new_other_income_kind').removeAttr('disabled');
   $("a.interaction-click-control-add-more").removeClass('hide');
   $("a[class*='income-edit'], a[class*='income-delete']").removeClass('disabled'); // legacy
+  $('#self-employed-income').removeClass('disabled');
+  $('a.new-income').removeClass('disabled');
+  $('a.self-employed-income-delete').removeClass('disabled');
```

Comment:
is this needed? this should be captured by the above a[class*='income-delete'] selector

Reply:
removed